### PR TITLE
Removed peer things and non-core things

### PR DESCRIPTION
**Pull request by rosshadden**
It turns out having these things as peer dependencies was messing things up for Erin, in that it was making her install ember things for backend projects.  I had gotten the peer dependency thing from airbnb's eslint config, but I guess it doesn't apply in the way I thought it did.

### CODE REVIEW

File: core.js [hunk 1]
```diff
@@ -1,8 +1,6 @@
 module.exports = {
 	env: {
-		'shared-node-browser': true,
```
Comment (rjhilgefort) [inline core.js]: shared node browser should stay so users can reliably use core

**Comment by rjhilgefort** [comment]
I don't think removing the peer dependency will cause the npm install to not fetch the ember eslint rules and what not. It will, however, suppress the error. Which is I guess what you're going for?

File: core.js (hunk 1 above)
Comment (rosshadden) [inline core.js]: Okay. I think `mongo`, `mocha`, etc. should all be in their own env files, but I agree this one should be there.

**Comment by rosshadden** [comment]
> I don't think removing the peer dependency will cause the npm install to not fetch the ember eslint rules and what not. It will, however, suppress the error. Which is I guess what you're going for?

Honestly, I don't know what the right thing to do is, or what the desired behavior is.  I saw that [airbnb uses peer dependencies](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/package.json#L58-L63) and kind of translated that to what I thought made sense in our project.  Do you think they should stay?

File: core.js (hunk 1 above)
Comment (rjhilgefort) [inline core.js]: completely agree, they should be moved over to the `frameworks` directory- if you feel like doing it in this mister. We could just leave them here for now and just make an issue to move and flesh them out at a later date.

**Comment by rjhilgefort** [comment]
I'd echo your thoughts about the peer dependency stuff, not sure what the right thing to do is. I'm fine removing.

File: core.js (hunk 1 above)
Comment (rosshadden) [inline core.js]: Sure, good idea.

File: environments/mongo.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
```
Comment (rjhilgefort) [inline environments/mongo.js]: Frameworks is admittedly a bad name, but this belongs there, and mixed into some other config that has the standard backend configuration.

File: package.json [hunk 3]
```diff
@@ -30,10 +30,7 @@
   },
   "peerDependencies": {
     "babel-eslint": "^6.0.4",
-    "eslint": "^2.7.0",
-    "eslint-plugin-ember": "^1.1.0",
-    "eslint-plugin-ember-cleanup": "^1.6.0",
-    "eslint-plugin-lodash": "^1.8.2"
+    "eslint": "^2.7.0"
```
Comment (rjhilgefort) [inline package.json]: leaving these couple peer deps?

Comment (rosshadden) [inline package.json]: Yeah, I think they make sense.  The problem with the plugin ones is they only conditionally make sense.

File: environments/mongo.js (hunk 2 above)
Comment (rosshadden) [inline environments/mongo.js]: `frameworks` => `apps`?

File: package.json (hunk 3 above)
Comment (rjhilgefort) [inline package.json]: 👍

File: environments/mongo.js (hunk 2 above)
Comment (rjhilgefort) [inline environments/mongo.js]: `modules`?

Comment (rosshadden) [inline environments/mongo.js]: I'm down for that.

**Comment by rjhilgefort** [comment]
@rosshadden This breaks config that depended on mocha being included in core. Please update and submit another PR.

**Comment by rjhilgefort** [comment]
I updated and pushed the tag

**Pull request merged**